DQM/HLTEvF/plugins/LumiMonitor.cc

```diff
@@ -0,0 +1,212 @@
+    numberOfPixelClustersVsLS_->setAxisTitle("number of pixel clusters",2);
+    
+    histname = "numberOfPixelClustersVsLumi"; histtitle = "number of pixel clusters vs lumi";
+    numberOfPixelClustersVsLumi_ = ibooker.bookProfile(histname, histtitle, 
+						       lumiPSet.getParameter<int32_t>("nbins"), lumiPSet.getParameter<double>("xmin"), lumiPSet.getParameter<double>("xmax"),
+						       pixelClusterPSet.getParameter<double>("xmin"), pixelClusterPSet.getParameter<double>("xmax"));
+    numberOfPixelClustersVsLumi_->setAxisTitle("(SCAL) inst lumi E30 [Hz cm^{-2}]",1);
+    numberOfPixelClustersVsLumi_->setAxisTitle("number of pixel clusters",2);
+
+    histname = "pixelLumiVsLS"; histtitle = "pixel-lumi vs LS";
+    pixelLumiVsLS_ = ibooker.bookProfile(histname, histtitle, 
+					 lsPSet.getParameter<int32_t>("nbins"), 0., double(lsPSet.getParameter<int32_t>("nbins")),
+					 pixellumiPSet.getParameter<double>("xmin"), pixellumiPSet.getParameter<double>("xmax"));
+    //    pixelLumiVsLS_->getTH1()->SetCanExtend(TH1::kAllAxes);
+    pixelLumiVsLS_->setAxisTitle("LS",1);
+    pixelLumiVsLS_->setAxisTitle("(pixel-lumi) inst lumi E30 [Hz cm^{-2}]",2);
+    
+    histname = "pixelLumiVsLumi"; histtitle = "pixel-lumi vs lumi";
+    pixelLumiVsLumi_ = ibooker.bookProfile(histname, histtitle, 
+					   lumiPSet.getParameter<int32_t>("nbins"), lumiPSet.getParameter<double>("xmin"), lumiPSet.getParameter<double>("xmax"),
+					   pixellumiPSet.getParameter<double>("xmin"), pixellumiPSet.getParameter<double>("xmax"));
+    pixelLumiVsLumi_->setAxisTitle("(SCAL) inst lumi E30 [Hz cm^{-2}]",1);
+    pixelLumiVsLumi_->setAxisTitle("(pixel-lumi) inst lumi E30 [Hz cm^{-2}]",2);
+  }
+
+  histname = "lumiVsLS"; histtitle = "lumi vs LS";
+  lumiVsLS_ = ibooker.bookProfile(histname, histtitle, 
+				  lsPSet.getParameter<int32_t>("nbins"), 0., double(lsPSet.getParameter<int32_t>("nbins")),
+				  lumiPSet.getParameter<double>("xmin"), lumiPSet.getParameter<double>("xmax"));
+  //  lumiVsLS_->getTH1()->SetCanExtend(TH1::kAllAxes);
+  lumiVsLS_->setAxisTitle("LS",1);
+  lumiVsLS_->setAxisTitle("(SCAL) inst lumi E30 [Hz cm^{-2}]",2);
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void LumiMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  //  int bx = iEvent.bunchCrossing();
+  int ls = iEvent.id().luminosityBlock();
+
+  float scal_lumi = -1.;
+  edm::Handle<LumiScalersCollection> lumiScalers;
+  iEvent.getByToken(lumiscalersToken_, lumiScalers);
+  if ( lumiScalers.isValid() && lumiScalers->size() ) {
+    LumiScalersCollection::const_iterator scalit = lumiScalers->begin();
+    scal_lumi = scalit->instantLumi();
+  } else {
+    scal_lumi = -1.;
+  }
+  lumiVsLS_ -> Fill(ls, scal_lumi);
```

Comment:
rather than filling with "-1" if there is no value from SCAL, why don't you simply skip the Fill ?
would there be any counter indication ?